runtime/vm/KeyHashTable.c

```diff
@@ -439,6 +441,27 @@ hashClassTableDelete(J9ClassLoader *classLoader, U_8 *className, UDATA className
 	return hashTableRemove(table, &key);
 }
 
+UDATA
+hashClassTablePackageDelete(J9VMThread *vmThread, J9ClassLoader* classLoader, J9ROMClass* romClass)
+{
+	if (isMHProxyPackage(romClass)) {
+		/* This generated package only exists for one hidden class and should be
+		* removed when its rom class is unloaded. hashClassTable package id's rely
+		* on there being a valid rom class.
+		*/
+		UDATA result = 0;
+		KeyHashTableClassEntry key;
+		key.tag = (UDATA)romClass | TAG_ROM_CLASS;
+		omrthread_monitor_enter(vmThread->javaVM->classTableMutex);
+		Trc_VM_hashClassTablePackageDelete(vmThread, romClass);
```

Comment:
Usually we put the event trace point after it happens, i,e, after the `hashTableRemove()` call.